bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/io/ChannelPool.java

```diff
@@ -62,8 +100,44 @@ public ChannelPool(Channel[] channels, List<HeaderInterceptor> headerInterceptor
     };
   }
 
+  private synchronized Channel getNextChannel() {
+    int currentRequestNum = requestCount.getAndIncrement();
+    int index = Math.abs(currentRequestNum % channels.length);
+    return channels[index];
+  }
+
   @Override
   public String authority() {
     return channels[0].authority();
   }
+
+  public synchronized WrappedChannel reserveChannel() {
+    Channel reserved;
+    boolean returned = false;
+    if (channels.length == 1) {
+      // Always leave one connection in the pool.
+      reserved = channels[0];
+      returned = true;
+    } else {
+      reserved = channels[channels.length - 1];
+      Channel[] newChannelArray = new Channel[channels.length - 1];
+      System.arraycopy(channels, 0, newChannelArray, 0, channels.length - 1);
```

Comment:
That might have an impact on the normal use case of ChannelPool.newCall().  We want that use case to be as fast as possible.